microsetta_interface/implementation.py

```diff
@@ -1316,6 +1324,27 @@ def get_create_nonhuman_source(*, account_id=None):
                                  account_id=account_id)
 
 
+# Note: ideally this would be represented as a DELETE, not as a POST
+# However, it is used as a form submission action, and HTML forms do not
+# support delete as an action
+def post_request_account_removal(*, account_id, body):
```

Comment:
When hitting this path on the Interface, I'm getting the following exception from Private API:
```
[2024-02-20 09:20:23 -0800] [19272] [ERROR] Exception on /api/accounts/9adc6ce5-c200-4d9e-aa52-df965f445304/removal_queue [PUT]
Traceback (most recent call last):
  File "/Users/api_user/miniconda3/envs/microsetta-private-api/lib/python3.8/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/api_user/miniconda3/envs/microsetta-private-api/lib/python3.8/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/api_user/miniconda3/envs/microsetta-private-api/lib/python3.8/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/api_user/miniconda3/envs/microsetta-private-api/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/api_user/miniconda3/envs/microsetta-private-api/lib/python3.8/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/api_user/miniconda3/envs/microsetta-private-api/lib/python3.8/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/api_user/miniconda3/envs/microsetta-private-api/lib/python3.8/site-packages/connexion/decorators/decorator.py", line 48, in wrapper
    response = function(request)
  File "/Users/api_user/miniconda3/envs/microsetta-private-api/lib/python3.8/site-packages/connexion/decorators/security.py", line 327, in wrapper
    return function(request)
  File "/Users/api_user/miniconda3/envs/microsetta-private-api/lib/python3.8/site-packages/connexion/decorators/uri_parsing.py", line 144, in wrapper
    response = function(request)
  File "/Users/api_user/miniconda3/envs/microsetta-private-api/lib/python3.8/site-packages/connexion/decorators/validation.py", line 384, in wrapper
    return function(request)
  File "/Users/api_user/miniconda3/envs/microsetta-private-api/lib/python3.8/site-packages/connexion/decorators/response.py", line 103, in wrapper
    response = function(request)
  File "/Users/api_user/miniconda3/envs/microsetta-private-api/lib/python3.8/site-packages/connexion/decorators/parameter.py", line 121, in wrapper
    return function(**kwargs)
  File "/Users/api_user/microsetta-private-api-master-overhaul/microsetta_private_api/api/_removal_queue.py", line 24, in request_remove_account
    rq_repo.request_remove_account(account_id, user_delete_reason)
  File "/Users/api_user/microsetta-private-api-master-overhaul/microsetta_private_api/repo/removal_queue_repo.py", line 65, in request_remove_account
    cur.execute(
psycopg2.errors.NotNullViolation: null value in column "id" violates not-null constraint
DETAIL:  Failing row contains (null, 9adc6ce5-c200-4d9e-aa52-df965f445304, 2024-02-20 09:20:23.823819-08, null).
```

Reply:
Can you provide me some steps to recreate this?

Reply:
All of this was on our staging server:
1) Deploy ayobi/microsetta-private-api:master_overhaul
2) Deploy ayobi/microsetta-interface:account_deletion_final
3) Create a new user account
4) When logged in as the new user, go to Account Details
5) Request Account Deletion
6) Enter "test" as the reason, then hit Ok

Is it working properly in your local environment?

Reply:
Yes, on my local environment I'm able to press the delete button and I get the 'We're sorry to see you go' page without any error and the account delete queue table is then populated.

Reply:
I'll investigate to see if I can recreate it somehow

Reply:
I'm not having issues with it using master either.

Reply:
So after digging into this, I've determined that the issue is that something on our staging server is losing the default value definition for the `id` column of that table, which is exceedingly weird. There are a couple of possibilities I'll need to explore before I can test this on staging.

Reply:
Very odd indeed, sounds good 👍

Reply:
@cassidysymons, that's quite weird -- any traction?

Reply:
@wasade yes, it was the issue with www-dev running Postgres 9.X rather than 13.X. I haven't had a chance to jump back into functional testing since Jeff upgraded the package on staging but I had isolated the issue to that before I contacted him.